Sources/MapboxNavigation/OverlayController.swift

```diff
@@ -0,0 +1,390 @@
+
+import UIKit
+import MapboxDirections
+import MapboxCoreNavigation
+import MapboxMaps
+import Turf
+import CoreLocation
+
+
+class OverlayController: NavigationComponent, ViewCycleObserver {
+    
+    // MARK: - Properties
+    
+    weak var navigationViewData: NavigationViewData!
+    
+    fileprivate var navigationView: NavigationView! {
+        return navigationViewData.navigationView
+    }
+    
+    fileprivate var navigationMapView: NavigationMapView {
+        return navigationViewData.navigationView.navigationMapView
+    }
+    
+    var detailedFeedbackEnabled: Bool = false
+    
+    var showsSpeedLimits: Bool = true {
+        didSet {
+            navigationView.speedLimitView.isAlwaysHidden = !showsSpeedLimits
+        }
+    }
+    
+    var floatingButtonsPosition: MapOrnamentPosition? {
+        get {
+            return navigationView.floatingButtonsPosition
+        }
+        set {
+            if let newPosition = newValue {
+                navigationView?.floatingButtonsPosition = newPosition
+            }
+        }
+    }
+    
+    var floatingButtons: [UIButton]? {
+        get {
+            return navigationView.floatingButtons
+        }
+        set {
+            navigationView.floatingButtons = newValue
+        }
+    }
+    
+    var reportButton: FloatingButton {
+        return navigationViewData.navigationView.reportButton
+    }
+    
+    typealias LabelRoadNameCompletionHandler = (_ defaultRoadNameAssigned: Bool) -> Void
+
+    var labelRoadNameCompletionHandler: (LabelRoadNameCompletionHandler)?
+    
+    // MARK: - Methods
+    
+    @objc func toggleMute(_ sender: UIButton) {
+        sender.isSelected = !sender.isSelected
+        
+        let muted = sender.isSelected
+        NavigationSettings.shared.voiceMuted = muted
+    }
+    
+    @objc func feedback(_ sender: Any) {
+        showFeedback()
+    }
```

Comment:
I guess these methods are only here because they also pertain to ornaments. I’m not sure they’d provide much value as part of a public OverlayController class, so it may not be worth the extra overhead to keep them separate from NavigationViewController.

Reply:
I think it does it's job to lighten the `NavigationViewController` for now. If we decide to expose `OrnamentsController` publicly, we'll need to do more tweaks and moving these methods should not become a problem if it will be needed